src/NzbDrone.Core/Download/DownloadService.cs

```diff
@@ -54,7 +60,26 @@ public void DownloadReport(RemoteEpisode remoteEpisode)
                 _rateLimitService.WaitAndPulse(uri.Host, TimeSpan.FromSeconds(2));
             }
 
-            var downloadClientId = downloadClient.Download(remoteEpisode);
+            string downloadClientId;
+            try
+            {
+                downloadClientId = downloadClient.Download(remoteEpisode);
+                _indexerStatusService.ReportSuccess(remoteEpisode.Release.IndexerId);
+            }
+            catch (ReleaseDownloadException ex)
+            {
+                var http429 = ex.InnerException as TooManyRequestsException;
```

Comment:
The original TooManyRequests is caught (all exceptions are caught) and thrown as ReleaseDownloadException. That's why I'm checking the InnerException.